mock_client.go

```diff
@@ -28,24 +38,27 @@ func NewMockClient(t *testing.T) *MockClient {
 }
 
 func (c *MockClient) Ping() error {
-	return c.Err
+	return c.PingErr
```

Comment:
can we use something like this here for backward compatibility and for cases where people don't want specific errors?
```suggestion
	return cmp.Or(c.PingErr, c.Err)
```

Reply:
Yea, that keeps backwards compatibility as well.